src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -19,6 +20,145 @@ namespace Microsoft.CodeAnalysis.CSharp
 {
     internal partial class Binder
     {
+#nullable enable
+
+        private enum EscapeLevel : uint
+        {
+            CallingMethod = Binder.CallingMethodScope,
+            ReturnOnly = Binder.ReturnOnlyScope,
+        }
+
+        /// <summary>
+        /// The destination in a method arguments must match (MAMM) check. This is 
+        /// created primarily for ref and out arguments of a ref struct. It also applies
+        /// to function pointer this and arglist arguments.
+        /// </summary>
+        private readonly struct MixableDestination
+        {
+            internal BoundExpression Argument { get; }
+
+            /// <summary>
+            /// In the case this is the argument for a ref / out parameter this will refer
+            /// to the corresponding parameter. This will be null in cases like arguments 
+            /// passed to an arglist.
+            /// </summary>
+            internal ParameterSymbol? Parameter { get; }
+
+            /// <summary>
+            /// This destination can only be written to by arguments that have an equal or
+            /// wider escape level. An destination that is <see cref="EscapeLevel.CallingMethod"/>
+            /// can never be written to by an argument that has a level of <see cref="EscapeLevel.ReturnOnly"/>.
+            /// </summary>
+            internal EscapeLevel EscapeLevel { get; }
+
+            internal MixableDestination(ParameterSymbol parameter, BoundExpression argument)
+            {
+                Debug.Assert(parameter.RefKind.IsWritableReference() && parameter.Type.IsRefLikeType);
```

Comment:
> && parameter.Type.IsRefLikeType [](http://example.com/codeflow?start=68&length=32)

It looks like this was dropped in commit 17. Was that intentional?

Reply:
Don't know what you mean, that check is in the code right now.

Reply:
Never mind, I misread the diff.